ee/debug/checkups/download_directory.go

```diff
@@ -28,21 +29,27 @@ func (c *downloadDirectory) Run(_ context.Context, extraFH io.Writer) error {
 		return nil
 	}
 
-	for _, downloadDir := range downloadDirs {
-		pattern := filepath.Join(downloadDir, "kolide-launcher*")
-		matches, err := filepath.Glob(pattern)
+	if extraFH != io.Discard {
+		zipWriter := zip.NewWriter(extraFH)
+		defer zipWriter.Close()
 
-		if err != nil {
-			fmt.Fprintf(extraFH, "Error listing files in directory (%s): %s\n", downloadDir, err)
-			continue
-		}
+		for _, downloadDir := range downloadDirs {
+			pattern := filepath.Join(downloadDir, "kolide-launcher*")
+			matches, err := filepath.Glob(pattern)
+			if err != nil {
+				continue
+			}
 
-		for _, match := range matches {
-			if info, err := os.Stat(match); err == nil {
-				c.files = append(c.files, fileInfo{
-					Name:    match,
-					ModTime: info.ModTime(),
-				})
+			for _, match := range matches {
+				if info, err := os.Stat(match); err == nil {
+					c.files = append(c.files, fileInfo{
+						Name:    match,
+						ModTime: info.ModTime(),
+					})
+					if err := addFileToZip(zipWriter, match); err != nil {
+						fmt.Fprintf(extraFH, "Error adding file to zip %s: %v\n", match, err)
+					}
```

Comment:
It looks like the update to this checkup will add the actual installer files to the flare, which I don't think we need -- I think this checkup was fine as it was before!

Reply:
Reverted